digicampipe/io/hessio_digicam.py

```diff
@@ -0,0 +1,297 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Components to read HESSIO data.
+
+This requires the hessio python library to be installed
+"""
+import logging
+
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+
+import digicampipe.utils as utils
+import numpy as np
+
+from digicampipe.io.containers_mc import DataContainer
+from ctapipe.core import Provenance
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription
+
+logger = logging.getLogger(__name__)
+
+try:
+    from pyhessio import open_hessio
+    from pyhessio import HessioError
+    from pyhessio import HessioTelescopeIndexError
+    from pyhessio import HessioGeneralError
+except ImportError as err:
+    logger.fatal(
+        "the `pyhessio` python module is required to access MC data: {}"
+        .format(err))
+    raise err
+
+__all__ = [
+    'hessio_event_source',
+]
+
+# import inspect
+# import pyhessio
+# print(inspect.getfile(pyhessio))
+
+
+def hessio_get_list_event_ids(url, max_events=None):
+    """
+    Faster method to get a list of all the event ids in the hessio file.
+    This list can also be used to find out the number of events that exist
+    in the file.
+
+    Parameters
+    ----------
+    url : str
+        path to file to open
+    max_events : int, optional
+        maximum number of events to read
+
+    Returns
+    -------
+    event_id_list : list[num_events]
+        A list with all the event ids that are in the file.
+
+    """
+    logger.warning("This method is slow. Need to find faster method.")
+    try:
+        with open_hessio(url) as pyhessio:
```

Comment:
Actually when the context "hessio" is already clear .. since we are in a file called "hessio_digicam" .. instead of "hessio_file" ... `file` would be good as well.

Reply:
I agree. I changed it and I will push the repaired code in a moment.